pkg/monitor/monitor/util.go

```diff
@@ -1215,16 +1180,16 @@ func defaultTidbMonitor(monitor *v1alpha1.TidbMonitor) {
 
 }
 
-func getMonitorStatefulSet(sa *core.ServiceAccount, secret *core.Secret, monitor *v1alpha1.TidbMonitor, tc *v1alpha1.TidbCluster, dc *v1alpha1.DMCluster, shard int32) (*apps.StatefulSet, error) {
+func getMonitorStatefulSet(sa *core.ServiceAccount, secret *core.Secret, monitor *v1alpha1.TidbMonitor, shard int32) (*apps.StatefulSet, error) {
 	statefulSet := getMonitorStatefulSetSkeleton(sa, monitor, shard)
-	initContainer := getMonitorInitContainer(monitor, tc)
+	initContainer := getMonitorInitContainer(monitor)
 	statefulSet.Spec.Template.Spec.InitContainers = append(statefulSet.Spec.Template.Spec.InitContainers, initContainer)
-	if dc != nil {
-		dmInitContainer := getMonitorDMInitContainer(monitor, dc)
+	if monitor.Spec.DM != nil {
```

Comment:
do we also need to check `monitor.Spec.DM.Clusters > 0`?

Reply:
When the number of dm clusters is zero, the `for loop` will not execute . WDYT? @csuzhangxc